src/main/java/org/opensearch/knn/index/IndexUtil.java

```diff
@@ -190,6 +190,14 @@ public static ValidationException validateKnnField(
             return exception;
         }
 
+        String vectorDataType = (String) fieldMap.get(VECTOR_DATA_TYPE_FIELD);
+        if (VectorDataType.BINARY.toString().equalsIgnoreCase(vectorDataType)) {
+            exception.addValidationError(
+                String.format("Field \"%s\" is of data type %s. Only FLOAT or BYTE is supported.", field, VectorDataType.BINARY)
```

Comment:
when using String.format set Locale.ROOT to ensure that UTs don't break if we validate the exception messages